src/network/networkd-dhcp-static-lease.c

```diff
@@ -0,0 +1,231 @@
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(userdata);
+
+        network = userdata;
+
+        r = lease_new_static(network, filename, section_line, &current_lease);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to create a new static lease: %s", rvalue);
+                return 0;
+        }
+
+        if (isempty(rvalue)) {
+                sd_dhcp_static_lease_unset_address(current_lease->static_lease);
+                return 0;
+        }
+
+        r = in_addr_from_string(AF_INET, rvalue, &addr);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to parse DHCPv4 IPv4 address data, ignoring static lease assignment: %s", rvalue);
+
+                return 0;
+        }
+
+        r = dhcp_static_lease_set_address(current_lease, &addr.in);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to set DHCPv4 IPv4 address data, ignoring static lease assignment: %s", rvalue);
+                return 0;
+        }
+
+        current_lease = NULL;
+
+        return 0;
+}
+
+int config_parse_dhcp_static_lease_hwaddr(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        Network *network;
+        _cleanup_(dhcp_static_lease_free_or_set_invalidp) DHCPStaticLease *current_lease = NULL;
+        struct ether_addr hwaddr;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(userdata);
+
+        network = userdata;
+
+        r = lease_new_static(network, filename, section_line, &current_lease);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to create a new static lease: %s", rvalue);
+                return 0;
+        }
+
+        if (isempty(rvalue)) {
+                sd_dhcp_static_lease_unset_client_id(current_lease->static_lease);
+                return 0;
+        }
+
+        r = ether_addr_from_string(rvalue, &hwaddr);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Not a valid MAC address, ignoring static lease assignment: %s", rvalue);
+                return 0;
+        }
+
+        r = dhcp_static_lease_set_client_id(current_lease, &hwaddr);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to set MAC address, ignoring static lease assignment: %s", rvalue);
+                return 0;
+        }
+
+        current_lease->section->invalid = false;
```

Comment:
Drop this line.

Reply:
If it were not for this line, this configuration:
```
[DHCPServerStaticLease]
MACAddress=
MACAddress=12:34:56:78:9a:bc
Address=10.1.1.2
```
would not result in a valid section, which it should as we discussed earlier. Whereas this configuration:
```
[DHCPServerStaticLease]
MACAddress=
Address=10.1.1.2
```
will and should be invalid.